src/normalize/text.ts

```diff
@@ -41,7 +41,27 @@ export default function normalizeText(editor: Editor, entry: NodeEntry): boolean
 
     if (Title.isTitle(node)) {
         // Title must be the first child of it's parent.
-        if (path[path.length - 1] > 0) {
+        // If it is found in the figure, move it.
+        // In other cases, change it to paragraph.
+        const [parent] = Editor.parent(editor, path)
+        const isFirst = path[path.length - 1] === 0
+        if (Figure.isFigure(parent)) {
+            // Figure must not have more than one title.
+            const [next, nextPath] = Editor.next(editor, { at: path }) ?? []
+            if (Title.isTitle(next)) {
+                Transforms.mergeNodes(editor, { at: nextPath })
+                return true
+            }
+
+            if (!isFirst) {
+                const newPath = [...path]
+                newPath[newPath.length - 1] = 0
+                Transforms.moveNodes(editor, { at: path, to: newPath })
+                return true
+            }
+        }
+
+        if (!isFirst) {
```

Comment:
I think it may be best to do this in general. In other words: if title is not the first child of its parent then:
1. merge it into the previous element if it is also a title;
2. otherwise change it into a paragraph.

Reply:
title w figurach w mikroekonomi jeżeli dobrze pamiętam są zazwyczaj ostatnim elementem, więc rozwalilibyśmy w ten sposób ich ładowanie

Reply:
> 1. merge it into the previous element if it is also a title;

It will not allow us to hit `enter` inside of the title in the section

> 2. otherwise change it into a paragraph.

Do you mean to remove the fragment about moving the title inside of the figure?
If so then it will break loading figures in microeconomy, because the title in them isn't the first child